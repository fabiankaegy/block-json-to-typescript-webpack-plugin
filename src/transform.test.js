const { printTypeDeclaration, createAttributesInterface, createContextInterface, createBlockInterface } = require('./transform');

describe('transform', () => {
	test('it works', () => {
		expect(1).toBe(1);
	});

	test('prints attribute type declaration', () => {
		const typeDeclaration = createAttributesInterface({
			attributes: {
				stringAttribute: {
					type: 'string',
				},
				numberAttribute: {
					type: 'number',
				},
				booleanAttribute: {
					type: 'boolean',
				},
				arrayAttribute: {
					type: 'array',
				},
				objectAttribute: {
					type: 'object',
				},
				randomAttribute: {
					type: 'random',
				},
				objectAttributeWithDefault: {
					type: 'object',
					default: {
						foo: 'bar',
					},
				},
				integerAttribute: {
					type: 'integer'
				}
			},
		}, 'TestAttributes');

		const typeDeclarationString = printTypeDeclaration(typeDeclaration);

		expect(typeDeclarationString).toMatchSnapshot();
	});

	test('prints block context type declaration', () => {
		const contextPropertyDeclaration = createContextInterface({}, 'TestContext');
		const typeDeclarationString = printTypeDeclaration(contextPropertyDeclaration);
		expect(typeDeclarationString).toMatchSnapshot();
	});

	test( 'prints block type declaration', () => {
		const blockDeclaration = createBlockInterface({}, 'TestBlock', {
			attributesInterfaceName: 'TestBlockAttributes',
			contextInterfaceName: 'TestBlockContext',
		});

		const typeDeclarationString = printTypeDeclaration(blockDeclaration);

		expect(typeDeclarationString).toMatchSnapshot();
	} );
});