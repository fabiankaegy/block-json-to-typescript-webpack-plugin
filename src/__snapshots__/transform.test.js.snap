// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform prints attribute type declaration 1`] = `
"interface TestAttributes {
    readonly stringAttribute?: string;
    readonly numberAttribute?: number;
    readonly booleanAttribute?: boolean;
    readonly arrayAttribute?: Array<any>;
    readonly objectAttribute?: Record<string, any>;
    readonly randomAttribute?: any;
    readonly objectAttributeWithDefault: Record<string, any>;
    readonly integerAttribute?: number;
    readonly style?: Record<string, any>;
}"
`;

exports[`transform prints block context type declaration 1`] = `
"interface TestContext {
}"
`;

exports[`transform prints block type declaration 1`] = `
"interface TestBlock {
    readonly name: string;
    readonly isSelected: boolean;
    readonly isSelectionEnabled: boolean;
    readonly clientId: string;
    readonly attributes: TestBlockAttributes;
    readonly context: undefined;
    setAttributes(attributes: Partial<TestBlockAttributes>): void;
    insertBlocksAfter(block: any): void;
    mergeBlocks(blocks: Array<any>): void;
    onRemove(callback: any): void;
    onReplace(callback: any): void;
    toggleSelection(): void;
    [key: string]: any;
}"
`;
